app/controllers/api/order_cycles_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  class OrderCyclesController < BaseController
+    include EnterprisesHelper
+    respond_to :json
+
+    skip_authorization_check
+
+    def products
+      products = ProductsRenderer.new(
+        distributor,
+        order_cycle,
+        customer,
+        params.slice(:q, :page, :per_page)
```

Comment:
Can we also sanitize `params[:q]` here? Not sanitizing might make this endpoint more powerful, but I think this search feature should only allow filters that we explicitly want to support.

Reply:
This deserves then a separate method. I was thinking about following the [strong params](https://github.com/rails/strong_parameters) convention and so we are a little closer but feel free to dismiss it.

Reply:
I've added the `strong_parameters` gem and used it here. :+1:

Reply:
Hmm... nope, I'll have to remove it. The gem was discontinued in 2014 and seems to have some issues. I don't think we can use it. :disappointed:

Reply:
@kristinalim I've added a new method for restricting the params here: https://github.com/openfoodfoundation/openfoodnetwork/pull/4345/commits/aa0217da08d65fbb4bfc7c9ab7909f749027dc1b

It's not very pretty. `#slice` doesn't take nested arguments. Let me know if there's a nicer way to do this!

Reply:
Yes, not very pretty. :wink: But I think that's okay for now, it's in a separate method and we can easily update when we have upgraded Rails.

Reply:
Yes, that was my point. Adding the gem it's too much. It was integrated into Rails 4 if I'm not mistaken. We can wait.